cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -45,12 +83,99 @@ class CoursewareSearchIndexer(object):
     Class to perform indexing for courseware search from different modulestores
     """
 
+    # Flags for where to index the information
+    ANALYSE = 1 << 0  # Add the information to the analysed content of the index
+    PROPERTY = 1 << 1  # Add the information as a property of the object being indexed (not analysed)
+
+    # List of properties to add to the index - each item in the list is a tuple of:
+    #   1) Property name to use
+    #   2) Where to add in the index (using flags above)
+    #   3) Whether the property is found in the 'about' area of the modulestore
+    #      (True) or is a property directly on the course object
```

Comment:
Can we turn this third parameter into a nicely-named/readable thing, versus just True/False?